src/zarr/core/array.py

```diff
@@ -3734,6 +3735,174 @@ class ShardsConfigParam(TypedDict):
 ShardsLike: TypeAlias = ChunkCoords | ShardsConfigParam | Literal["auto"]
 
 
+async def from_array(
+    data: Array,
```

Comment:
```suggestion
    data: Array | npt.ArrayLike,
```
As discussed, this function should also work with numpy arrays.